include/kernel/thread.h

```diff
@@ -280,20 +280,31 @@ static inline bool_t checkBudget(void)
     return false;
 }
 
-/* Everything checkBudget does, but also set the thread
- * state to ThreadState_Restart. To be called from kernel entries
- * where the operation should be restarted once the current thread
- * has budget again.
- */
 
-static inline bool_t checkBudgetRestart(void)
+/* This is called from kernel entries where operation are done only if the
+ * thread still has budget. Otherwise it needs to be restarted first.
+ */
+static inline bool_t checkBudgetWithRestart(void)
 {
+    /* The assumption here is, that an operation was started by the current
+     * thread, thus this thread must be runnable. Otherwise something is wrong
+     * or the assumption does not hold and the function was called in a wrong
+     * context. It's not fatal if this assert is removed in release builds,
```

Comment:
The "it's not fatal" comment is true of all asserts? They are mostly documentation as well as an aid to anyone developing new features (break this assert and you will have a very bad time).

Happy to improve the docs, but please keep them short and avoid documenting obvious facts about asserts.